src/components/modals/SupportRequestModal.js

```diff
@@ -0,0 +1,194 @@
+      return prev;
+    });
+  };
+
+  emailChangeHandler = (e) => {
+    let emailText = e.target.value;
+    this.state.vaild = /.+@.+\.[A-Za-z]+$/.test(emailText) ? true: false; 
+    this.setState(prev => {
+      prev.email = emailText;
+      return prev;
+    });
+  };
+
+  render() {
+
+    return (
+      SupportRequestBaseModal({
+        id: "SupportRequestModal",
+        disableOkBtn: (this.state.name === '' || this.state.email === '' || this.state.subject === '' || this.state.description === '' || this.state.vaild === false),
+        showModal: this.props.showModal,
+        onRequestClose: this.closeHandler,
+        onAfterOpen: this.afterOpenHandler,
+        imgSrc: "/images/icon_add_help.png",
+        color: "common",
+        title: "Contact Us",
+        height: this.state.height,
+        top: this.state.top,
+        action: { label: "Submit", handler: this.OKHandler },
+      },
+
+        [
+          form({ className: "form-horizontal css-form", name: "consentForm", noValidate: true, encType: "multipart/form-data" }, [
+            !this.state.isLogged && div({ className: "form-group first-form-group" }, [
+              label({ id: "lbl_name", className: "common-color" }, ["Name *"]),
+              input({ id: "txt_name", placeholder:'What should we call you?', value: this.state.name, className: "form-control col-lg-12", onChange: this.nameChangeHandler, required: true }),
+            ]),
+            div({ className: "form-group first-form-group" }, [
+              label({ id: "lbl_type", className: "common-color" }, ["Type *"]),
+              select({id: "txt_question", className: "col-lg-12 select-wrapper", value: this.state.type, onChange: this.typeChangeHandler, required: true}, [
+                option({ value: "question"} , ["Question"]),
+                option({ value: "bug"}, ["Bug"]),
+                option({ value: "feature_request"}, ["Feature Request"])
+              ]),  
+            ]),
+            div({ className: "form-group first-form-group" }, [
+              label({ id: "lbl_description", className: "common-color" }, ["How can we help you" + this.state.first_name + "? *"]),
+              input({ id: "txt_subject", placeholder:'Enter a subject', rows: "5", className: "form-control col-lg-12 vote-input", onChange: this.subjectChangeHandler, required: true }),
+              textarea({ id: "txt_description", placeholder:'Enter a description', rows: "5", className: "form-control col-lg-12 vote-input", onChange: this.descriptionChangeHandler, required: true }),
+            ]),
+            div({ className: "form-group first-form-group" }, [
+              label({ id: "lbl_attachment", className: "common-color" }, ["Attachment"]),
+              input({ type: "file", id: "txt_attachment", placeholder: 'Attach a file?', className: "form-control col-lg-12 vote-input common-color", onChange: this.attachmentChangeHandler, ref: 'fileUpload', required: false }),
```

Comment:
TOL (thinking out loud) ... I think this should be triggered on submit, not change. We could chain the file upload to happen before the submit so the order of operations is maintained.

More TOL, this only allows for a single file upload which is fine. I can see the need for multiples, but not for this story ... possibly a future enhancement.